File: flutter_Backend/CreateApi/src/main/java/com/shohab/CreateApi/model/Salary.java
package com.shohab.CreateApi.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.sql.Date;

@Data // Generates getters, setters, toString, equals, and hashCode methods
@NoArgsConstructor // Generates a no-argument constructor
@Entity
@AllArgsConstructor
public class Salary {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    private Employee employee;
    private BigDecimal empBonus;
    private BigDecimal empAdvance;
    private BigDecimal amount;
    private BigDecimal TotalAmount;
    private BigDecimal totalSalary;
    private String date;
    private String empworkingdays;
    private String empPresent;
    private String empLeaves;
    private String empabsent;
    private String empFines;
    private String workingDays;

    public Employee getEmployee() {
        return employee;
    }

    public Salary(Employee employee) {
        this.employee = employee;
    }

    public void setEmployee(Employee employee) {
        this.employee = employee;
    }
}

File: PayrollProjectNew/payroll_app/lib/src/features/salary/model/salary.dart
import 'package:payroll_app/src/features/employee/model/employee.dart';

class Salary {
  int? id;
  Employee? employee;
  double? empBonus;
  double? empAdvance;
  double? amount;
  double? totalSalary;
  String? date;
  double? totalAmount;

  Salary(
      {this.id,
        this.employee,
        this.empBonus,
        this.empAdvance,
        this.amount,
        this.totalSalary,
        this.date,
        this.totalAmount});

  Salary.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    employee = json['employee'] != null
        ? Employee.fromJson(json['employee'])
        : null;
    empBonus = json['empBonus'];
    empAdvance = json['empAdvance'];
    amount = json['amount'];
    totalSalary = json['totalSalary'];
    date = json['date'];
    totalAmount = json['totalAmount'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    if (this.employee != null) {
      data['employee'] = this.employee!.toJson();
    }
    data['empBonus'] = this.empBonus;
    data['empAdvance'] = this.empAdvance;
    data['amount'] = this.amount;
    data['totalSalary'] = this.totalSalary;
    data['date'] = this.date;
    data['totalAmount'] = this.totalAmount;
    return data;
  }
}